src/Microsoft.AspNetCore.Antiforgery/Internal/DefaultAntiforgeryTokenStore.cs

```diff
@@ -69,22 +69,14 @@ public void SaveCookieToken(HttpContext httpContext, string token)
             Debug.Assert(httpContext != null);
             Debug.Assert(token != null);
 
-            var options = new CookieOptions
-            {
-                HttpOnly = true,
-#pragma warning disable 618
-                Domain = _options.CookieDomain,
-#pragma warning restore 618
-                SameSite = SameSiteMode.Strict,
-                Secure = _options.RequireSsl
-            };
-
-#pragma warning disable 618
-            if (_options.CookiePath != null)
+            var options = _options.Cookie.Build(httpContext);
+
+            options.Secure = options.Secure || _options.RequireSsl;
```

Comment:
This has historically been here along with `RequireSsl`. I'm not really sure if the reasons are important or not.